hkmc2/shared/src/main/scala/hkmc2/semantics/Elaborator.scala

```diff
@@ -1580,12 +1585,10 @@ extends Importer:
   class VarianceTraverser(var changed: Bool = true) extends Traverser:
     override def traverseType(pol: Pol)(trm: Term): Unit = trm match
       case Term.TyApp(lhs, targs) =>
-        lhs.symbol.flatMap(sym => sym.asTpe orElse sym.asMod orElse sym.asObj) match
+        lhs.symbol.flatMap(sym => sym.asTpe) match
           case S(sym: ClassSymbol) =>
             sym.defn match
             case S(td: ClassDef) =>
-              if td.tparams.sizeCompare(targs) =/= 0 then
-                raise(ErrorReport(msg"Wrong number of type arguments" -> trm.toLoc :: Nil)) // TODO BE
```

Comment:
Because of the change above, the checks here started to report false
errors (which was due to the change of the order of `asTpe`, `asMod`,
etc.) 

I think it's not appropriate to do the checks here because it doesn't
have enough information to distinguish overloaded definitions, and we already
have the same check in the resolver. So I just removed them.